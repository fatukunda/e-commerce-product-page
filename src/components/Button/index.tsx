import React, { FunctionComponent } from "react";
import { ButtonType, ButtonSize } from "src/theme";

export interface ButtonProps {
  children: JSX.Element | JSX.Element[];
  handleClick: () => void;
  size: string;
  type: string;
}

const Button: FunctionComponent<ButtonProps> = ({
  children,
  handleClick,
  size,
  type,
}): JSX.Element => {
  const classNames = `${ButtonType[type]} ${ButtonSize[size]}`
  return (
    <button
      type="button"
      className={classNames}
      onClick={handleClick}
      data-testid="main-btn"
    >
      {children}
    </button>
  );
};

export default Button;
