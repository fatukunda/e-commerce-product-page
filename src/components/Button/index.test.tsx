import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Button from "./";

const renderButton = () =>
  render(
    <Button>
      <span>Click me!</span>
    </Button>
  );
describe('Tests for <Button/>', () => {
    afterEach(cleanup)

    it('renders without crashing', () => {
        renderButton()
    })
})

test("Button: Snapshot", () => {
  const component = renderer.create(
    <Button>
      <span>Test</span>
    </Button>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
