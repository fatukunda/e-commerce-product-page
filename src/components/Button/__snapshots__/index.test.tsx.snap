// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button: Snapshot 1`] = `
<button
  className="flex w-full items-center justify-center gap-2 rounded-lg bg-primary p-4 font-bold text-white shadow-lg shadow-primary/50 transition-all duration-300 hover:bg-primary/50"
  data-testid="main-btn"
  type="button"
>
  <span>
    Test
  </span>
</button>
`;
